commit 5aad39dbb324aad4cac3f44967ee1fc1b3d2a168
Author: Aaron Seigo <aseigo@kde.org>
Date:   Sun Nov 20 18:50:41 2011 +0100

    don't count custom panel contaiments (e.g. the grouping desktop) the same as a desktop containment
    
    from the "useful things missing from bug reports" and "wondering wtf is wrong with people
    while trying to get the useful bits out of a bug report" files ...
    
    BUG:260360

diff --git a/plasma/containment.cpp b/plasma/containment.cpp
index 00a8a77..9e3a505 100644
--- a/plasma/containment.cpp
+++ b/plasma/containment.cpp
@@ -987,7 +987,9 @@ void ContainmentPrivate::setScreen(int newScreen, int newDesktop, bool preventIn
     //kDebug() << activity() << "setting screen to " << newScreen << newDesktop << "and type is" << type;
 
     Containment *swapScreensWith(0);
-    if (type == Containment::DesktopContainment || type >= Containment::CustomContainment) {
+    const bool isDesktopContainment = type == Containment::DesktopContainment ||
+                                      type == Containment::CustomContainment;
+    if (isDesktopContainment) {
         // we want to listen to changes in work area if our screen changes
         if (toolBox) {
             if (screen < 0 && newScreen > -1) {
@@ -1011,8 +1013,7 @@ void ContainmentPrivate::setScreen(int newScreen, int newDesktop, bool preventIn
         }
     }
 
-    if (newScreen < numScreens && newScreen > -1 &&
-        (type == Containment::DesktopContainment || type >= Containment::CustomContainment)) {
+    if (newScreen < numScreens && newScreen > -1 && isDesktopContainment) {
         q->resize(corona->screenGeometry(newScreen).size());
     }
 
